pkg/expression/vs_helper.go

```diff
@@ -0,0 +1,98 @@
+// Copyright 2024 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package expression
+
+import (
+	"strings"
+
+	"github.com/pingcap/errors"
+	"github.com/pingcap/tidb/pkg/parser/ast"
+	"github.com/pingcap/tidb/pkg/parser/model"
+	"github.com/pingcap/tidb/pkg/parser/mysql"
+	"github.com/pingcap/tidb/pkg/types"
+)
+
+var (
+	vsDistanceFnNamesLower = map[string]struct{}{
+		strings.ToLower(ast.VecL1Distance):           {},
+		strings.ToLower(ast.VecL2Distance):           {},
+		strings.ToLower(ast.VecCosineDistance):       {},
+		strings.ToLower(ast.VecNegativeInnerProduct): {},
+	}
+)
+
+// VectorSearchExpr defines a minimal Vector Search expression, which is
+// a vector distance function, a column to search with, and a reference vector.
+type VectorSearchExpr struct {
+	DistanceFnName model.CIStr
+	Vec            types.VectorFloat32
+	Column         *Column
+}
+
+// ExtractVectorSearch extracts a VectorSearchExpr from an expression.
+// NOTE: not all VectorSearch functions are supported by the index. The caller
+// needs to check the distance function name.
+func ExtractVectorSearch(expr Expression) (*VectorSearchExpr, error) {
```

Comment:
why not let VectorSearchExpr be an another expression Expression implementor

Reply:
then we could do this in getFunction verify args

and the handling in `pkg/planner/core/task.go` xx31 will be more clean